molvis/gaussian/python/processors/cubesourcemolvis.py

```diff
@@ -0,0 +1,155 @@
+ # ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ # WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ # DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ # ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ # (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ # LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ # ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ # (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ # SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ #
+ #################################################################################
+ 
+
+import inviwopy as ivw
+import ivwdataframe as df
+import ivwmolvis
+import gaussianutil
+
+from pathlib import Path
+
+class CubeSourceMolvis(ivw.Processor):
+    def __init__(self, id, name):
+        ivw.Processor.__init__(self, id, name)
+
+        self.volumeOutport = ivw.data.VolumeOutport("chargedensity")
+        self.addOutport(self.volumeOutport, owner=False)
+
+        self.meshOutport = ivw.data.MeshOutport("atoms")
+        self.addOutport(self.meshOutport)
+
+        self.dataframeOutport = df.DataFrameOutport("atomInformation")
+        self.addOutport(self.dataframeOutport)
+
+        self.addOutport(ivwmolvis.MolecularStructureOutport("molecule"))
+
+        self.cubeFilePath = ivw.properties.FileProperty("cube", "cube")
+        self.addProperty(self.cubeFilePath)
+
+        self.dataRange = ivw.properties.DoubleMinMaxProperty(
+            "dataRange", "Data Range", 0.0, 1.0, -1.70e308, 1.79e308)
+        self.addProperty(self.dataRange)
+        self.dataRange.semantics = ivw.properties.PropertySemantics("Text")
+        self.dataRange.readOnly = True
+
+        self.useCustomRange = ivw.properties.BoolProperty(
+            "useCustomRange", "Overwrite Data Range", False)
+        self.addProperty(self.useCustomRange)
+
+        self.customDataRange = ivw.properties.DoubleMinMaxProperty(
+            "customDataRange", "Custom Data Range", 0.0, 1.0, -1.70e308, 1.79e308)
+        self.addProperty(self.customDataRange)
+        self.properties.customDataRange.semantics = ivw.properties.PropertySemantics(
+            "Text")
+
+        self.flipSign = ivw.properties.BoolProperty(
+            "flipSign", "Flip Sign of Charge", False)
+        self.addProperty(self.flipSign)
+
+        self.centerData = ivw.properties.BoolProperty(
+            "centerData", "Center Data", True)
+        self.addProperty(self.centerData)
+
+        self.pm = ivw.PickingMapper(self, 1, lambda x: self.callback(x))
+
+    @staticmethod
+    def processorInfo():
+        return ivw.ProcessorInfo(
+    		classIdentifier = "org.inviwo.CubeSourceMolvis", 
+    		displayName = "Cube Source Molvis",
```

Comment:
Remove Molvis from name? Possibly add something like Molecules or ChargeDensityField if it makes sense.

Reply:
There is already a regular `CubeSource` python processor (same module & folder, see https://github.com/inviwo/modules/blob/master/molvis/gaussian/python/processors/cubesource.py). This one here outputs the atom positions as molecular data structure instead of a regular Inviwo Mesh.